test/unit/utils/OrderedMsgChain.test.ts

```diff
@@ -165,168 +226,227 @@ describe('OrderedMsgChain', () => {
+        it('does not hold onto old markMessageExplicitly messages', (done) => {
+            const msgs: StreamMessage[] = []
+            const gapHandler = jest.fn((_from, to) => {
+                if (to.timestamp === 2) {
+                    util.markMessageExplicitly(msg2)
+                    util.markMessageExplicitly(msg4)
+                }
+            })
 
-    it('does not hold onto old markMessageExplicitly messages', (done) => {
-        const msgs: StreamMessage[] = []
-        const gapHandler = jest.fn((from, to) => {
-            if (to.timestamp === 2) {
-                util.markMessageExplicitly(msg2)
-                util.markMessageExplicitly(msg4)
-            }
+            util = new OrderedMsgChain('publisherId', 'msgChainId', (msg: StreamMessage) => {
+                msgs.push(msg)
+                if (msgs[msgs.length - 1].getMessageRef().timestamp === 5) {
+                    util.markMessageExplicitly(msg2)
+                    assert.deepStrictEqual(msgs, [msg1, msg3, msg5]) // msg 2 & 3 will be missing
+                    assert.deepStrictEqual([
+                        util.markedExplicitly.has(msg1),
+                        util.markedExplicitly.has(msg2),
+                        util.markedExplicitly.has(msg3),
+                        util.markedExplicitly.has(msg4),
+                        util.markedExplicitly.has(msg5),
+                    ], [
+                        false,
+                        false,
+                        false,
+                        false,
+                        false,
+                    ])
+                    expect(gapHandler).toHaveBeenCalledTimes(1)
+                    done()
+                }
+            }, gapHandler, 100, 100)
+
+            util.on('error', done)
+
+            util.add(msg1)
+            // missing msg2
+            util.add(msg3)
+            // missing msg4
+            util.add(msg5)
         })
 
-        util = new OrderedMsgChain('publisherId', 'msgChainId', (msg: StreamMessage) => {
-            msgs.push(msg)
-            if (msgs[msgs.length - 1].getMessageRef().timestamp === 5) {
-                util.markMessageExplicitly(msg2)
-                assert.deepStrictEqual(msgs, [msg1, msg3, msg5]) // msg 2 & 3 will be missing
-                assert.deepStrictEqual([
-                    util.markedExplicitly.has(msg1),
-                    util.markedExplicitly.has(msg2),
-                    util.markedExplicitly.has(msg3),
-                    util.markedExplicitly.has(msg4),
-                    util.markedExplicitly.has(msg5),
-                ], [
-                    false,
-                    false,
-                    false,
-                    false,
-                    false,
-                ])
-                expect(gapHandler).toHaveBeenCalledTimes(1)
-                done()
-            }
-        }, gapHandler, 100, 100)
+        it('still requests gapfill for gaps after msg marked explicitly', (done) => {
```

Comment:
I'll add a comment but the problem was this:

> Marking a message anywhere inside a gap would skip straight to the next message in the queue, rather than trying to fill the gap between the marked message and the next queued message.

Previously it wouldn't try fill a gap between a marked message and the next queued message, it'd just skip the entire gap the moment it got any marked message inside the gap

Reply:
Added a comment.